tomodachi/opentelemetry/logging.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, Any, Dict, Optional, cast
+
+from opentelemetry import trace
+from opentelemetry._logs import get_logger, get_logger_provider
+from opentelemetry.sdk._logs import LoggerProvider, LoggingHandler
+from opentelemetry.util.types import Attributes
+from tomodachi.__version__ import __version__ as tomodachi_version
+
+if TYPE_CHECKING:  # pragma: no cover
+    try:
+        from structlog.typing import EventDict, WrappedLogger
+    except (ImportError, ModuleNotFoundError):
+        from structlog.types import EventDict, WrappedLogger
+
+
+def add_trace_structlog_processor(logger: WrappedLogger, method_name: str, event_dict: EventDict) -> EventDict:
+    span = trace.get_current_span()
+    if not span.is_recording():
+        return event_dict
+
+    ctx = span.get_span_context()
+
+    if ctx.span_id is not None:
+        event_dict["span_id"] = hex(ctx.span_id)
+    if ctx.trace_id is not None:
+        event_dict["trace_id"] = hex(ctx.trace_id)
```

Comment:
Added `parent_span_id` to the structlog output at least to provide some extra possibility of tracing using those logs. However, the `parent_span_id` will only makes sense if there was actually a log record emitted during the parent span as well, otherwise it'll be missing and the tracing is broken any way.

Reply:
Right! Very good point!